dz-02/src/solutions.js

```diff
@@ -0,0 +1,185 @@
+    return parseInt(a) + parseInt(b);
+  },
+
+  /**
+   * Returns formatted string for the given date.
+   * Format should be `{day}-{month}-{fullYear}` (all numbers).
+   * @param {Date} date
+   */
+  formatDate(date) {
+    return `${date.getDate()}-${date.getMonth() + 1}-${date.getFullYear()}`;
+  },
+
+  /**
+   * Sorts the given `numberArray` in ascending order.
+   * Use array `.sort` method. Sort is done in place so there is no need to return anything.
+   * @param {number[]} numberArray
+   */
+  sortNumberArray(numberArray) {
+    numberArray.sort((a, b) => a - b);
+  },
+
+  /**
+   * Multiplies all the elements in the array by 2 _in place_
+   * (edits the given array) and returns it.
+   * @param {number[]} numberArray
+   */
+  multiplyArrayByTwo(numberArray) {
+    for (let i = 0; i < numberArray.length; i++) {
+      numberArray[i] *= 2;
+    }
+    return numberArray;
+  },
+
+  /**
+   * Multiplies all the elements in the array by 2 and returns them
+   * in a new array.
+   * @param numberArray
+   */
+  multiplyArrayByTwoNew(numberArray) {
+    result = numberArray.map(item => item * 2);
+    return result;
+  },
+
+  /**
+   * Create two classes: `Person` and `Programmer`. `Programmer` class extends `Person`.
+   * Person class has `name` property (set via constructor) and `getName` method (calls `callGetName` with name`).
+   * Programmer class has `language` property provided to constructor (and `name` inherited from `Person`) and `getLanguage` method (calls `callGetLanguage` with `language`)
+   * Return object with created classes, `return { Person, Programmer }`.
+   *
+   * NOTE: class methods should use `bind`, function expression syntax won't work here because code isn't transpiled.
+   *
+   * @param {Function} callGetName
+   * @param {Function} callGetLanguage
+   */
+
+  classInheritance(callGetName, callGetLanguage) {
+    class Person {
+      constructor(name) {
+        this.name = name;
+      }
+      getName() {
+        callGetName(this.name);
+      }
+    }
+    class Programmer extends Person {
+      constructor(name, language) {
+        super(name);
+        this.language = language;
+      }
+      getLanguage() {
+        callGetLanguage(this.language);
+      }
+    }
+    return { Person, Programmer };
+  },
+
+  /**
+   * EXTRA CREDIT TASK -> Closure trick with async. Async is not important here and has nothing to do with the solution.
+   *
+   * **This is variant of probably most common "big firm" interview question with closures.**
+   *
+   * This task has more easier solutions (e.g. using `let` instead of `var`), but desired solutions included Closures.
+   *
+   * Call the `consumer` function once every second three times giving it loop iterator as argument.
+   * Use the provided for loop, do not change for loop, but feel free to modify setTimeout.
+   * @param {Function} consumer
+   */
+  timeoutIncrement(consumer) {
+    for (var i = 1; i <= 3; i += 1) {
+      (a =>
```

Comment:
🏅 bravo

Idealno rjesenje je factory funkcija koja vraca funkciju koja se onda predaje `setTimeoutu`, ali i tvoje rjesenje je na istom tragu samo je lagano drugacije u runtimeu. Budemo to komentirali malo na predavanju iducem jer je to savrseni uvod u event-loop 😄